### PR TITLE
[dcl.attr] Introduce 'Recommended practice' paragraphs.

### LINKED ISSUE #2978: [dcl.attr.*] and Recommended Practice
One of the tools the C standard uses for its attribute wording is a Recommended Practice section (http://www.open-std.org/jtc1/sc22/wg14/www/docs/n2385.pdf 6.7.11). These sections correspond to the same wording we put in informative notes currently.

From what I understand, a Recommended Practice is a special kind of section (like Constraints or Semantics) that places normative encouragement on implementations without being a normative requirement. Would it make sense to follow suit in C++? I think it may allow us to speak more authoritatively than a note while still serving the purpose of not being a normative requirement.

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #2978.

### CODE REVIEW

**Comment by zygoloid** [comment]
@jensmaurer Please can you mail the CWG reflector just to draw people's attention to this change so that this style can be used going forward?

**Comment by zygoloid** [comment]
Actually... let's get CWG to look over this before applying it. Converting "may" to "should", while not introducing a requirement, does introduce a recommendation, and that seems like something for which we should ask for oversight.

**Comment by jensmaurer** [comment]
Reference to CWG e-mail: http://lists.isocpp.org/core/2020/01/8153.php

File: source/intro.tex [hunk 1]
```diff
@@ -600,6 +600,10 @@
 \Cpp{} and C\@. Certain features of \Cpp{} exist solely for compatibility
 purposes; \ref{depr} describes those features.
 
+\pnum
+Throughout this document,
+non-normative encouragement for implementations is introduced by \recommended.
```
Comment (tvaneerd) [inline source/intro.tex]: That sounds pretty definitive.  Are you implying *all* instances of non-normative encouragement? What about all the places we missed, and are still in notes or wherever?

**Comment by zygoloid** [comment]
Approved by CWG 2020-02-14.

**Pull request merged**